activerecord/test/cases/relation/where_clause_test.rb

```diff
@@ -181,6 +181,39 @@ class WhereClauseTest < ActiveRecord::TestCase
       assert_equal WhereClause.empty, WhereClause.empty.or(where_clause)
     end
 
+    test "or places common conditions before the OR" do
+      wcs = (0..6).map do |i|
+        WhereClause.new([table["col_#{i}"].eq(bind_param(i))])
```

Comment:
This sort of thing makes this test really difficult to read. How about something like this?

```ruby
a = WhereClause.new(
  [table["id"].eq(bind_param(1)), table["name"].eq(bind_param("Sean"))],
)
b = WhereClause.new(
  [table["id"].eq(bind_param(1)) ,table["hair_color"].eq(bind_param("black"))],
)
common = WhereClause.new(
  [table["id"].eq(bind_param(1))],
)
or_clause = WhereClause.new(
  [table["name"].eq(bind_param("Sean")).or(table["hair_color"].eq(bind_param("black")))],
)

assert_equal common + or_clause, a.or(b)
```